airbyte-integrations/connectors/source-stripe/integration_tests/abnormal_state.json

```diff
@@ -236,5 +222,61 @@
       "stream_state": { "created": 10000000000 },
       "stream_descriptor": { "name": "transactions" }
     }
+  },
+  {
+    "type": "STREAM",
+    "stream": {
+      "stream_state": { "updated": 10000000000 },
+      "stream_descriptor": { "name": "application_fees_refunds" }
```

Comment:
You're right. But I left those as is intentionally, to make sure things work as expected for both new and legacy cursor fields as we have to support them both and we do support them in the code. If you think we'd better migrate them all, I won't have objections

Reply:
Why do we need to support both if the migration asks for a schema refresh [here](https://github.com/airbytehq/airbyte/pull/29330/files#diff-00b4e0295734534266dbff8cd901255c963e4e6fa37d8ce09a4bb4caf230d858R6)? Maybe I'm confused by what "smooth migration" and "not smooth migration" is

Reply:
Because the state of the sync persists with legacy cursor field inside. So when running sync for the first time after rolling out this change, we'll have a legacy state, but will produce new state with the new cursor field

Reply:
I don't think a schema refresh performs a state migrations, does it?

Reply:
You are right. So basically, even though we ask the user to perform something, a reset would be too disruptive. Got it!